### PR TITLE
configure s3 backend with state locking on dynamodb

### PR DESCRIPTION
**Pull request by petrolifero**

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    aws = {
+      source  = "hashicorp/aws"
+      version = "3.40.0"
+    }
+  }
+  backend "s3" {
+    bucket = "terraformstateinfrarepository"
+    key    = "infraRepository/state"
+    region = "sa-east-1"
+    dynamodb_table = "terraformstatelocking"
```
Comment (dbofmmbt) [inline main.tf]: The values on lines 9 and 12 can have underline (_) betweens word? Would be easier to read. Or camel case, if you prefer.

Comment (petrolifero) [reply main.tf]: s3 don't allow _, so I change with -

**Comment by eduardocanellas** [comment]
Looks good to me!

File: main.tf (hunk 1 above)
Comment (dbofmmbt) [reply main.tf]: Looks good to me!

**Review by dbofmmbt — APPROVED** [review]

**Pull request merged**